### ISSUE TITLE
Preference to auto-select "In Selection" when something is selected

### ISSUE THREAD

**Issue by vr8hub** [coteditor/CotEditor#823]
The editor I've used for 15+ years will auto-check the "In Selection" checkbox in Find if more than one line is currently selected (note, more than one line, not just any selection at all). This makes it a *lot* easier to do searches in a selection, as we don't have to remember to check and uncheck the box.

It would be very helpful if CotEditor had a Preference that would enable that same behavior.

**Comment by 1024jp** [comment]
Thank you for the suggestion. I feel it would be useful. I'll consider and implement as long as there are no other conflicts.

**Comment by 1024jp** [comment]
I considered implementing this behavior but reached a decision not to adopt, because such an implicit behavior can make users confuse especially when the search text itself contains some line breaks.

**Comment by vr8hub** [comment]
That implicit behavior is used by many other editors on the market. Users are not confused. They are helped, by not having to constantly remember to check and check a box that can be done far more easily, and efficiently, by the computer.

I'm not sure what line breaks being contained in the search text itself has to do with anything.

If the highlighted text is more than one line, then it's highlighted because we want to act on it. Thus, if we do a search (or search/replace), we want it to act on that highlighted text. That's why we highlighted it before we did the search.
If the highlighted text is only on one line, then we want to search on that highlighted text.

That is perfectly clear behavior. It is easily understandable, it is easily explainable, and it does what the user wants to happen 99.99% of the time. The other .01% the user can override it.
